sdk/communication/AzureCommunicationChat/Source/ChatThreadClient.swift

```diff
@@ -0,0 +1,336 @@
+            switch result {
+            case .success:
+                completionHandler(.success(()), httpResponse)
+
+            case let .failure(error):
+                completionHandler(.failure(error), httpResponse)
+            }
+        }
+    }
+
+    /// Sends a message to a thread.
+    /// - Parameters:
+    ///    - chatMessage : Request for sending a message.
+    ///    - options: A list of options for the operation.
+    ///    - completionHandler: A completion handler that receives a status code on success.
+    public func sendMessage(
+        request: SendChatMessageRequest,
+        withOptions options: SendChatMessageOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<SendChatMessageResult>
+    ) {
+        client.send(chatMessage: request, chatThreadId: threadId, withOptions: options) { result, httpResponse in
+            switch result {
+            case let .success(sendMessageResult):
+                completionHandler(.success(sendMessageResult), httpResponse)
+
+            case let .failure(error):
+                completionHandler(.failure(error), httpResponse)
+            }
+        }
+    }
+
+    /// Gets a message by id.
+    /// - Parameters:
+    ///    - chatMessageId : The message id.
+    ///    - options: Get chat message options
+    ///    - completionHandler: A completion handler that receives the chat message on success.
+    public func getMessage(
+        chatMessageId: String,
+        withOptions options: GetChatMessageOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<ChatMessage>
+    ) {
+        client
+            .getChatMessage(
+                chatThreadId: threadId,
+                chatMessageId: chatMessageId,
+                withOptions: options
+            ) { result, httpResponse in
+                switch result {
+                case let .success(chatMessage):
+                    completionHandler(.success(chatMessage), httpResponse)
+
+                case let .failure(error):
+                    completionHandler(.failure(error), httpResponse)
+                }
+            }
+    }
+
+    /// Updates a message.
+    /// - Parameters:
+    ///    - request: Request to update the message.
+    ///    - chatMessageId : The message id.
+    ///    - options: Update chat message options
+    ///    - completionHandler: A completion handler that receives a status code on success.
+    public func updateMessage(
+        request: UpdateChatMessageRequest,
+        chatMessageId: String,
+        withOptions options: UpdateChatMessageOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<Void>
+    ) {
+        client
+            .update(
+                chatMessage: request,
+                chatThreadId: threadId,
+                chatMessageId: chatMessageId,
+                withOptions: options
+            ) { result, httpResponse in
```

Comment:
Mirror:
```swift
public func update(message: UpdateChatMessageRequest, withOptions:...)
```

Alternatively, if `UpdateChatMessageRequest` just contains a `ChatMessage` object, consider accepting that for the convenience layer and wrapping all of those types. This comment would apply throughout this PR.